rustic-cargo.el

```diff
@@ -481,16 +481,18 @@ If running with prefix command `C-u', read whole command from minibuffer."
     (rustic-run-cargo-command command (list :mode 'rustic-cargo-run-mode))))
 
 (defun rustic-cargo-run-get-relative-example-name ()
-  "Run 'cargo run --example' if current buffer within a 'exmaples' directory."
-  (if rustic--buffer-workspace
-      (let ((relative-filenames
-             (split-string (file-relative-name buffer-file-name rustic--buffer-workspace) "/")))
-        (if (string= "examples" (car relative-filenames))
-            (let ((size (length relative-filenames)))
-              (cond ((eq size 2) (file-name-sans-extension(nth 1 relative-filenames))) ;; examples/single-example1.rs
-                    ((> size 2) (car (nthcdr (- size 2) relative-filenames))) ;; examples/example2/main.rs
-                    (t nil))) nil))
-    nil))
+  "Run 'cargo run --example' if current buffer within a 'exmaples' directory.
+In rust, examples normally runs within single crate or a project member within workspaces."
+  (let* ((buffer-project-root (rustic-buffer-crate))
```

Comment:
The name `buffer-project-root` seems incorrect when we call `rustic-buffer-crate` ?

Reply:
Fixed, see ⬇️, the `rustic-buffer-crate` happened works when it's just like `examples/example1.rs`